### PR TITLE
Add new puppetdb-plugin

### PR DESCRIPTION
**Pull request by tdabasinskas**
## PuppetDB frontend plugin (`@backstage/plugin-puppetdb`)

Building on the functionality introduced under #16184, this new frontend plugin allows viewing PuppetDB reports (including logs and events in them) for Resource entities that have `puppet.com/certname` annotation set.

#### Main page with the reports list:

Reports list

#### Events for the specific report:

Events

#### Logs of the specific report:

Logs

### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/itchy-bulldogs-hunt.md
Comment (awanlin) [inline .changeset/itchy-bulldogs-hunt.md:5]: Can you include a link to the plugin's README? These changesets turn into release notes so it would make it easier for people ti dive right into this if they are interested

File: microsite/data/plugins/puppetdb.yaml
Comment (awanlin) [inline microsite/data/plugins/puppetdb.yaml:8]: Just so we have it some where but where does this come from? It does include the Puppet logo and I'm not sure the rules on it's usage. We might need to change this

File: plugins/puppetdb/README.md
Comment (awanlin) [inline plugins/puppetdb/README.md:7]: The general practice is to have a `/docs/images` folder and put these there, feel free to follow that. You also have some other example in the PR description that would be great to see here. These screenshots really help give people an idea of the features and are super helpful.

File: plugins/puppetdb/README.md
Comment (awanlin) [inline plugins/puppetdb/README.md:3]: Am I understanding this correctly: that without using the `PuppetDbEntityProvider` this plugin won't work? If that is true then I might want to call that out in bold or make that more obvious

File: plugins/puppetdb/README.md
Comment (awanlin) [inline plugins/puppetdb/README.md:35]: My Puppet experience is limited and we used Foreman so I may be mixing things up but should we include a link to documentation about authenticating against the API? Is it normal to have this wide open? I guess I would expect to seem some auth header with a token of some sort in this example.

**Review by awanlin — CHANGES_REQUESTED** [review]
Another great contribution @tdabasinskas. Just left some comments about the changeset and README.

File: plugins/puppetdb/README.md
Comment (awanlin) [reply plugins/puppetdb/README.md:7]: Actually looks like you have them in the `/assets` folder, that should work fine too. Would still be great to see them in the README

File: plugins/puppetdb/README.md
Comment (tdabasinskas) [reply plugins/puppetdb/README.md:3]: The requirement is to have `puppet.com/certname` annotation set for Resource kind of entities. This is done automatically by PuppetDbEntityProvider, but it can be done in some other way as well.

I've updated the README to include a new `Prerequisites` section and have elaborated more on the need of this annotation. Hope it makes it more clear.

File: microsite/data/plugins/puppetdb.yaml
Comment (tdabasinskas) [reply microsite/data/plugins/puppetdb.yaml:8]: I've updated it to point to the one listed on Wikipedia under Creative Commons - https://commons.wikimedia.org/wiki/File:Puppet_transparent_logo.svg. Should it be OK?

File: plugins/puppetdb/README.md
Comment (tdabasinskas) [reply plugins/puppetdb/README.md:35]: PuppetDB supports only certificate based or no authentication at all. Certificate based authentication might be tricky to implement here, I guess, so I would like to leave if for the future if anyone needs it.

For now, I've added the following to the README:
> The PuppetDB [should be configured](https://www.puppet.com/docs/puppetdb/7/configure.html#host) to allow being accessed from your Backstage instance.

**Comment by tdabasinskas** [comment]
Thanks for so prompt feedbac, @awanlin! I've pushed some changes addressing the review 😊

File: microsite/data/plugins/puppetdb.yaml
Comment (awanlin) [reply microsite/data/plugins/puppetdb.yaml]: Yeah, I'll be honest I don't know. I just mention it as it's come up in the past. I'll leave this to the maintainers to decide on.

Comment (awanlin) [reply microsite/data/plugins/puppetdb.yaml]: But you can put the image in `/microsite/static/img`

File: plugins/puppetdb/README.md
Comment (awanlin) [reply plugins/puppetdb/README.md:40]: OK, that's fine with me. 👍

File: plugins/puppetdb/README.md
Comment (awanlin) [inline plugins/puppetdb/README.md:1]: Can you add the screenshots? Sorry, I feel like I left a comment about this but it's gone now 🤔 

The screenshots really help to give people a feel for the features. You could create a Features section like the Azure DevOps plugin does to showcase them - https://github.com/backstage/backstage/tree/master/plugins/azure-devops

Comment (awanlin) [reply plugins/puppetdb/README.md:1]: Sorry, I think I was just expecting them at the top and didn't scroll down. Thanks!

**Review by awanlin — APPROVED** [review]
Alright, LGTM but we'll need to maintainers to do their review 🚀

**Review by jhaals — CHANGES_REQUESTED** [review]
Thanks for the PR! I've made an initial pass with some suggestions :)

File: microsite/data/plugins/puppetdb.yaml
Comment (tdabasinskas) [reply microsite/data/plugins/puppetdb.yaml]: Added the image to the mentioned path.

File: plugins/puppetdb/src/api/types.ts
Comment (tdabasinskas) [reply plugins/puppetdb/src/api/types.ts:177]: Thanks. I'll have this in mind if we will decide to export this API later on.

**Comment by tdabasinskas** [comment]
Thanks for the review, @jhaals. I've addressed the comments.

File: microsite/data/plugins/puppetdb.yaml
Comment (tdabasinskas) [reply microsite/data/plugins/puppetdb.yaml:8]: Any recommendations on how to better deal with this? Does the same concern apply of using externally hosted image?

**Pull request merged**
